### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by Tentanus**
The basic setup for the Parser.
It checks for a few things allready [see top get_data.c]

TO DO:
- [ ] - more error checking does still need to be added
- [ ] - validation for map (flood fill from Solypsist)

### CODE REVIEW

File: src/parser/get_data.c [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+#include "libft.h"
+#include <sys/types.h>
+
+static const char *g_id_str[TYPE_ID_MAX] = {
+	[TYPE_ID_NORTH] = "NO", [TYPE_ID_SOUTH] = "SO", [TYPE_ID_WEST] = "WE",
+	[TYPE_ID_EAST] = "EA",	[TYPE_ID_FLOOR] = "F",	[TYPE_ID_CEILING] = "C"};
+
+// clang-format off
+/*
```
Comment (Tentanus) [inline src/parser/get_data.c]: Change to enum ERR_NUM

File: src/parser/get_data.c [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+ *			 		get_texture:	after path line isn't empty [ERR_PARSE_FORMAT]
+ *
+ *			i		get_colour:		colour value exceeds limits [ERR_PARSE_RGB]
+ *			,		get_colour:		formatting error (255,255,255) [ERR_PARSE_FORMAT]
+ *
+ *			F		check_filled:	Checks if a value has allready been set [ERR_PARSE_FILLED]
+ */
+// clang-format on
+
+void get_error(char err)
```
Comment (Tentanus) [inline src/parser/get_data.c]: make sure static functions are static

File: lib/libft
Comment (Tentanus) [inline lib/libft:1]: make libft Norm

File: src/parser/get_data.c [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	else if (id == TYPE_ID_CEILING)
+		return (ft_ternary(info->col_ce != 0, FAILURE, SUCCESS));
+	return (FAILURE);
+}
+
+bool set_infovalue(t_cub3d *info, t_type_id id, char *str)
+{
+	char err;
+
+	err = '\0';
```
Comment (Tentanus) [inline src/parser/get_data.c]: add ERR_ENUM here

**Review by Tentanus — COMMENTED** [review]
Some minor ToDo's

- [ ]  Make Norminette

**Comment by Tentanus** [comment]
@Solypsist Please review

**Review by Solypsist — APPROVED** [review]

**Review by Solypsist — APPROVED** [review]

**Pull request merged**